### PR TITLE
Bug with regression models for the multivariate targets fixed

### PR DESCRIPTION
**Pull request by nicl-nno**
Current problem: when placing some models for time series forecasting in PrimaryNode, an error occurs in the model's fit stage (for forecast_length > 1).

List of models that fails: xgbreg, adareg, gbr, svr, sgdr
List of models that work correctly: linear, lasso, ridge, knnreg, dtreg, treg, rfr

The bug fixed by with involvement  of MultiOutputRegressor that wraps the models that can not produce multivariate prediction directly (see https://stackoverflow.com/questions/39540123/muti-output-regression-in-xgboost).

Additional:

- multivariate target disable if it not necessary (to speed-up the fit of the ts models)
- ts compsing example extended

### CODE REVIEW

File: fedot/core/models/evaluation/evaluation.py [hunk 1]
```diff
@@ -147,7 +149,23 @@ def fit(self, train_data: InputData):
...
+                if train_data.task.task_type == TaskTypesEnum.classification:
+                    multiout_func = MultiOutputClassifier
+                elif train_data.task.task_type in \
+                        [TaskTypesEnum.regression, TaskTypesEnum.ts_forecasting]:
+                    multiout_func = MultiOutputRegressor
+                else:
+                    raise ex
+                # apply MultiOutput
+                sklearn_model = multiout_func(sklearn_model)
+                sklearn_model.fit(train_data.features, train_data.target)
```
Comment (BarabanovaIrina) [inline fedot/core/models/evaluation/evaluation.py:166]: Может вынести в вспомогательную функцию с говорящим названием?

File: test/tasks/test_forecasting.py [hunk 2]
```diff
@@ -150,7 +158,7 @@ def test_regression_chain_forecast_multistep_correct():
     chain.fit(input_data=train_data)
     _, rmse_on_test, _, _ = get_rmse_value(chain, train_data, test_data)
 
-    rmse_threshold = np.std(test_data.target)
+    rmse_threshold = np.std(test_data.target) * 1.3
```
Comment (BarabanovaIrina) [inline test/tasks/test_forecasting.py:161]: магическое число

File: fedot/core/chains/ts_chain.py [hunk 3]
```diff
@@ -25,13 +25,17 @@ def forecast(self, initial_data: InputData, supplementary_data: InputData) -> Ou
         if supplementary_data.task.task_type is not TaskTypesEnum.ts_forecasting:
             raise ValueError('TsForecastingChain can be used for the ts_forecasting task only.')
 
+        forecast_length = supplementary_data.task.task_params.forecast_length
+
         supplementary_data_for_forecast = copy(supplementary_data)
         supplementary_data_for_forecast.task.task_params.make_future_prediction = True
+        # if forecast_length > 1:
```
Comment (BarabanovaIrina) [inline fedot/core/chains/ts_chain.py:32]: Не совсем понятный комментарий, потому что по факту проверки не происходит, код выполнится в любом случае. Возникает вопрос, что будет, если forecast_length <1?(если логически это вообще возможно)

**Review by BarabanovaIrina — CHANGES_REQUESTED** [review]

File: fedot/core/composer/gp_composer/gp_composer.py [hunk 4]
```diff
@@ -23,6 +23,12 @@
     RegressionMetricsEnum
 from fedot.core.repository.tasks import Task, TaskTypesEnum
 
+sample_split_ration_for_tasks = {
+    TaskTypesEnum.classification: 0.8,
+    TaskTypesEnum.regression: 0.8,
+    TaskTypesEnum.ts_forecasting: 0.5
```
[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/core/composer/gp_composer/gp_composer.py:29]: А для clustering вообще не нужно разделять train и test в ходе подбора модели, поэтому его и нет.

File: fedot/core/data/preprocessing.py [hunk 5]
```diff
@@ -44,17 +44,27 @@ def apply(self, data):
 
 
 class Normalization(PreprocessingStrategy):
-    def __init__(self):
-        self.default = ImputationStrategy()
+    def __init__(self, with_imputation=True):
```
[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/core/data/preprocessing.py]: Сделал проще - [унаследовал](https://github.com/nccr-itmo/FEDOT/pull/222/files#diff-9186fa38f252fd01248620f72f94bcc5c668b6dae4a528aa779dacca6268039fR88-R100) Normalization и NormalizationWithImputation от соответсвующих Scaling и ScalingWithImputation , чтобы избежать дублирования. 

В идеале конечно они должно от единого класса наследоваться вообще (что-то вроде SklearnPreprocessingStrategy) или же иметь название scaler-а как гиперпараметр.

File: fedot/core/models/evaluation/evaluation.py (hunk 1 above)
Comment (nicl-nno) [reply fedot/core/models/evaluation/evaluation.py]: [Вынес](https://github.com/nccr-itmo/FEDOT/pull/222/files#diff-9b69b7629ae678324e9459f501e8b8d1d478e427cade16c1f3403b285b92cc57R291-R302).

File: test/tasks/test_forecasting.py (hunk 2 above)
Comment (nicl-nno) [reply test/tasks/test_forecasting.py]: [Вынес](https://github.com/nccr-itmo/FEDOT/pull/222/files#diff-468bb001a0e75e989084579f4e04ba5ac7e27a43ec5c2ce9260e48724b63fddcR21-R23) в функцию

File: fedot/core/chains/ts_chain.py (hunk 3 above)
Comment (nicl-nno) [reply fedot/core/chains/ts_chain.py]: Убрал вообще этот комментарий как устаревшей.

На минимальный forecast_length добавил [проверку](https://github.com/nccr-itmo/FEDOT/pull/222/files#diff-99771fd9ad61e2e454012b0e0904b1c22a8f6c90bb22b6649f7123f0c4dbb784R19-R21) и [тест](https://github.com/nccr-itmo/FEDOT/pull/222/files#diff-468bb001a0e75e989084579f4e04ba5ac7e27a43ec5c2ce9260e48724b63fddcR289-R292).

File: fedot/core/data/preprocessing.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (BarabanovaIrina) [reply fedot/core/data/preprocessing.py]: 1. А поправь, пожалуйста еще вот [тут](https://github.com/nccr-itmo/FEDOT/blob/0006347a7d2575c1997a666fa6b1678ec2755db5/fedot/core/data/preprocessing.py#L172). Там должно быть ScalingWithImputation. И можно сразу добавить новые.

2. А так и задумано, что теперь нет preprocessing.normalize(), а вместо этого scaler = MinMaxScaler()? они одинаково работают? Просто если логика действий не нарушается, то ок, что такое наследование. А если да, то со стороны выглядит странно, что Normalizaiton наследуется от Scaling. (Хотя может я не сильно в понятиях шарю).

[the remark this answers is not in the document]
Comment (nicl-nno) [reply fedot/core/data/preprocessing.py]: 1. [Добавил](https://github.com/nccr-itmo/FEDOT/pull/222/commits/c0fa0372637b90d5651ad21e7d96367e22096086#diff-9186fa38f252fd01248620f72f94bcc5c668b6dae4a528aa779dacca6268039fR168-R172).

2. Это @Dreamlone вроде менял - там проблема в том, что normalize на практике как-то оч криво работает, хоть его суть сводится как раз к MinMaxScaler.

Но в любом случае можно и scaler = preprocessing.normalize() при необходимости написать. С порядком наследования наверное нет особого смысла сейчас разбираться, все равно это вроде планировали рефактировать.

[the remark this answers is not in the document]
Comment (BarabanovaIrina) [reply fedot/core/data/preprocessing.py]: ок

**Review by BarabanovaIrina — APPROVED** [review]
